src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -1246,17 +1261,24 @@ class MacroAssembler: public Assembler {
   // invalidated, so there will be a trap at its start.
   // For this to work, the destination address in the trampoline is
   // always updated, even if we're not using the trampoline.
+  // --
 
   // Emit a direct call if the entry address will always be in range,
-  // otherwise a trampoline call.
+  // otherwise a patchable far call.
   // Supported entry.rspec():
   // - relocInfo::runtime_call_type
   // - relocInfo::opt_virtual_call_type
   // - relocInfo::static_call_type
   // - relocInfo::virtual_call_type
   //
   // Return: the call PC or null if CodeCache is full.
+  address patchable_far_call(Address entry) {
```

Comment:
For runtime_call_type, I don't think we ever update the target/destination, so the name "patchable" seems not quite right for them.  Also, for runtime_call_type, since they never change, we can decide early if a near call is possible when the destination is always reachable (based on the bounds of code cache), which is what aarch64 does for trampoline_call.

Reply:
Yes. My thinking was, the site is still patachable, even if some sites don't need that capability.
The reason why this patch ignores near calls is because the short reach of JAL +-1MB (so normally only a few stubs can be reach from a few nmethods).
But it is on the enhancement list.

I don't mind changing the name, feel free to suggest something!

Reply:
The key things seems to be that they are typed with a relocInfo, so maybe `reloc_call`?

Reply:
Ok, fixed.